packages/snap/src/utils/fetchIpfsContent.ts

```diff
@@ -0,0 +1,68 @@
+import type { ProviderMeta } from '@galactica-net/galactica-types';
+
+const ipfsPatterns = [
+  /^ipfs:\/\/([^/?#]+)/iu,
+  /^https?:\/\/[^/]+\/ipfs\/([^/?#]+)/iu,
+  /^\/ipfs\/([^/?#]+)/iu,
+];
+
+const extractContentId = (ipfsUrl: string) => {
```

Comment:
Can you please add a couple unit tests to make sure that those functions and patterns work?